tests/run/i17255/J.java

```diff
@@ -0,0 +1,13 @@
+package p;
+
+public class J {
+	public static J j = new J();
+	public static p.J f() {
+		return p.J.j;
+	}
+	public static p.Module$ module() {
```

Comment:
This ends up being incorrectly typed as `p.Module` (i.e. the class) rather than `p.Module.type`

Reply:
When the tree reaches TASTy, this looks like
```
    87:       DEFDEF(12) 16 [module]
    90:         EMPTYCLAUSE
    91:         SELECTtpt 17 [Module$] // should be SELECTtpt 19 [Module[ModuleClass]]
    93:           SHAREDtype 3
    95:         ELIDED
    96:           TYPEREF 19 [Module[ModuleClass]]
    98:             SHAREDtype 3
   100:         STATIC
```

Reply:
can leave this to follow-up, as it is still better than status quo. However this does rely on the compiler (aka all TASTy clients) intrinsically knowing how to resolve `Module$` when the definition is actually `Module[ModuleClass]`

Reply:
new issue https://github.com/lampepfl/dotty/issues/19806

Reply:
Thank you for reviewing. I will investigate #19806 too.

Reply:
I'm not familiar with TASTy, so it may take a while...

Reply:
The way to solve this is in `typedSelect` replacing the name in the `Select` tree, like you did in `javaFindMember`

Reply:
How can I test TASTy?

Reply:
you can do it manually with the `-Yprint-tasty` flag which will print any generated TASTy after pickler phase, make sure you use the required `-Yjava-tasty` also so that you generate tasty for java sources